clap_complete/src/dynamic/completer.rs

```diff
@@ -76,8 +76,61 @@ pub fn complete(
         } else if arg.is_escape() {
             is_escaped = true;
             state = ParseState::ValueDone;
-        } else if let Some(_long) = arg.to_long() {
-        } else if let Some(_short) = arg.to_short() {
+        } else if let Some((flag, value)) = arg.to_long() {
+            if let Ok(flag) = flag {
+                state = if let Some(opt) = current_cmd.get_arguments().find(|a| {
+                    a.get_long_and_visible_aliases()
+                        .map_or(false, |v| v.into_iter().find(|s| *s == flag).is_some())
+                }) {
+                    match opt.get_action() {
+                        clap::ArgAction::Set | clap::ArgAction::Append => {
+                            if value.is_some() {
+                                ParseState::ValueDone
+                            } else {
+                                ParseState::Opt(opt.clone())
+                            }
+                        }
+                        _ => {
+                            if value.is_some() {
+                                ParseState::ValueDone
+                            } else {
+                                ParseState::ValueDone
+                            }
```

Comment:
Please list out all of the known `ArgAction` states to make the intent more clear for what all is `ValueDone`